core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/PreAggregateCaseAggregations.java

```diff
@@ -405,14 +416,22 @@ private Optional<CaseAggregation> extractCaseAggregation(Symbol aggregationSymbo
             }
         }
 
+        Expression cumulativeAggregationDefaultValue;
+        if (caseExpression.defaultValue().type().equals(aggregationType)) {
+            cumulativeAggregationDefaultValue = caseExpression.defaultValue();
```

Comment:
Is this needed in order for `newAssignments.getMap().values().equals(projectAssignments.getMap().values())` to work correctly?
I would add a comment explaining the need for the `if`.